sunpy/coordinates/ephemeris.py

```diff
@@ -192,24 +192,30 @@ def get_horizons_coord(body, time='now', id_type='majorbody'):
         >>> _ = c.__exit__()
     """
     obstime = parse_time(time)
+    array_time = np.reshape(obstime, (-1,))  # Convert to an array, even if scalar
 
     # Import here so that astroquery is not a module-level dependency
     from astroquery.jplhorizons import Horizons
     query = Horizons(id=body, id_type=id_type,
                      location='500@10',      # Heliocentric (mean ecliptic)
-                     epochs=obstime.tdb.jd)  # Time must be provided in JD TDB
+                     epochs=array_time.tdb.jd.tolist())  # Time must be provided in JD TDB
     try:
         result = query.vectors()
     except Exception:  # Catch and re-raise all exceptions, and also provide query URL if generated
         if query.uri is not None:
             log.error(f"See the raw output from the JPL HORIZONS query at {query.uri}")
         raise
     log.info(f"Obtained JPL HORIZONS location for {result[0]['targetname']}")
+    log.debug(f"See the raw output from the JPL HORIZONS query at {query.uri}")
 
-    vector = CartesianRepresentation(result[0]['x', 'y', 'z'])*u.AU
+    # JPL HORIZONS results are sorted by observation time, so this sorting needs to be undone
+    unsorted_indices = obstime.argsort().argsort()  # Stacked call obtains the unsorted order
```

Comment:
this comment would be better if it described how calling argsort twice achieves this. I hope from your variable name we know what it's trying to do :wink:

Reply:
Okay, expansive comment added